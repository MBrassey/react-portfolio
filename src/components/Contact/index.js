import { validateEmail } from "../../utils/helpers";
import React, { useState } from "react";

function Contact() {
  const [formState, setFormState] = useState({
    name: "",
    email: "",
    message: "",
  });
  const { name, email, message } = formState;
  const [errorMessage, setErrorMessage] = useState("");

  function handleChange(e) {
    if (e.target.name === "email") {
      const isValid = validateEmail(e.target.value);
      console.log(isValid);
      // Validate user input
      if (!isValid) {
        setErrorMessage("Your email is invalid.");
      } else {
        if (!e.target.value.length) {
          setErrorMessage(`${e.target.name} is required.`);
        } else {
          setErrorMessage("");
        }
      }
    }
    if (!errorMessage) {
      setFormState({ ...formState, [e.target.name]: e.target.value });
    }
  }

  function handleSubmit(e) {
    e.preventDefault();
    console.log(formState);
  }

  return (
    <div class="contact">
      <section>
        <form id="contact-form" onSubmit={handleSubmit}>
          <div>
            <label htmlFor="name">Name:</label>
            <input
              type="text"
              name="name"
              defaultValue={name}
              onBlur={handleChange}
            />
          </div>
          <div>
            <label htmlFor="email">Email address:</label>
            <input
              type="email"
              name="email"
              defaultValue={email}
              onBlur={handleChange}
            />
          </div>
          <div>
            <label htmlFor="message">Message:</label>
            <textarea
              name="message"
              rows="5"
              defaultValue={message}
              onBlur={handleChange}
            />
          </div>
          {errorMessage && (
            <div>
              <p className="error-text">{errorMessage}</p>
            </div>
          )}
          <button type="submit">Submit</button>
        </form>
      </section>
    </div>
  );
}

export default Contact;
