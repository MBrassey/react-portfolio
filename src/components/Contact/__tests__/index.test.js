import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';

import Contact from '..';

afterEach(cleanup);

describe('Contact component', () => {
  it('renders', () => {
    render(<Contact />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Contact />);

    expect(asFragment()).toMatchSnapshot();
  });
})
