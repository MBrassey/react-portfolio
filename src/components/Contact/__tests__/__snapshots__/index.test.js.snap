// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <div
        class="contact"
      >
        <section>
          <form
            id="contact-form"
          >
            <div>
              <label
                for="name"
              >
                Name:
              </label>
              <input
                name="name"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="email"
              >
                Email address:
              </label>
              <input
                name="email"
                type="email"
                value=""
              />
            </div>
            <div>
              <label
                for="message"
              >
                Message:
              </label>
              <textarea
                name="message"
                rows="5"
              />
            </div>
            <button
              type="submit"
            >
              Submit
            </button>
          </form>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
